src/webgpu/api/validation/vertex_state.spec.ts

```diff
@@ -303,73 +310,169 @@ g.test('vertex_attribute_shaderLocation_limit')
+    } else {
+      attributesB.push({ format: 'float', offset: 0, shaderLocation: shaderLocationB });
+    }
+
+    // Use the attributes to make the list of vertex buffers not that we might be setting the same vertex
+    // buffer twice but that only happens when it is the only vertex buffer.
+    const vertexBuffers: GPUVertexBufferLayoutDescriptor[] = [];
+    vertexBuffers[vertexBufferIndexA] = { arrayStride: 256, attributes: attributesA };
+    vertexBuffers[vertexBufferIndexB] = { arrayStride: 256, attributes: attributesB };
+
+    const success = shaderLocationA !== shaderLocationB;
+    t.testVertexState(success, { vertexBuffers });
+  });
+
+g.test('vertex_shader_input_location_limit')
+  .desc(
+    `Test that vertex shader's input's location decoration must be less than maxVertexAttributes.
+   - Test for shaderLocation 0, 1, limit - 1, limit`
+  )
+  .subcases(() =>
+    params().combine(
+      poptions('testLocation', [0, 1, kMaxVertexAttributes - 1, kMaxVertexAttributes])
+    )
+  )
+  .fn(t => {
+    const { testLocation } = t.params;
+
+    const shader = t.generateTestVertexShader([
       {
-        arrayStride: 2 * SIZEOF_FLOAT,
+        type: 'vec4<f32>',
+        location: testLocation,
+      },
+    ]);
+
+    const vertexBuffers: GPUVertexBufferLayoutDescriptor[] = [
+      {
+        arrayStride: 512,
         attributes: [
           {
             format: 'float',
             offset: 0,
-            shaderLocation: 0,
-          },
-          {
-            format: 'float',
-            offset: 0,
-            shaderLocation: 1,
+            shaderLocation: testLocation,
           },
         ],
       },
-    ],
-  };
-  {
-    // Control case: pipeline with one input per attribute
-    const code = `
-      [[location(0)]] var<in> a : vec4<f32>;
-      [[location(1)]] var<in> b : vec4<f32>;
+    ];
 
-      [[builtin(position)]] var<out> Position : vec4<f32>;
-      [[stage(vertex)]] fn main() -> void {
-        Position = vec4<f32>(0.0, 0.0, 0.0, 0.0);
-        return;
-      }
-  `;
-    const descriptor = t.getDescriptor(vertexState, code);
-    t.device.createRenderPipeline(descriptor);
-  }
-  {
-    // Check it is valid for the pipeline to use a subset of the VertexState
-    const code = `
-      [[location(0)]] var<in> a : vec4<f32>;
+    const success = testLocation < kMaxVertexAttributes;
+    t.testVertexState(success, { vertexBuffers }, shader);
+  });
 
-      [[builtin(position)]] var<out> Position : vec4<f32>;
-      [[stage(vertex)]] fn main() -> void {
-        Position = vec4<f32>(0.0, 0.0, 0.0, 0.0);
-        return;
-      }
-    `;
-    const descriptor = t.getDescriptor(vertexState, code);
-    t.device.createRenderPipeline(descriptor);
-  }
-  {
-    // Check for an error when the pipeline uses an attribute not in the vertex input
-    const code = `
-      [[location(2)]] var<in> a : vec4<f32>;
+g.test('vertex_shader_input_location_in_vertex_state')
+  .desc(``)
```

Comment:
should have at least a short description

Reply:
Whoops :)